### PR TITLE
[Provider] Namecheap.com

### PR DESCRIPTION
**Pull request by coltonehrman**
- Added provider code for namecheap.com

Reminder: Namecheap API user must add server IP to whitelist

### CODE REVIEW

File: src/providers/goDaddy.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ const findKey = (key: string): string => {
 
 export const getDomains = async (): Promise<Provider> => {
   const providerKeys = getKeys()
+
+  console.log(findKey(keys[1]))
```
Comment (aortizoj15) [inline src/providers/goDaddy.ts:26]: remove console.log

File: src/api/admin.ts [hunk 2]
```diff
@@ -167,6 +167,7 @@ app.patch('/providerKeys/:id', (req, res) => {
 })
 
 app.get('/providers', async (_, res) => {
+  console.log(providers)
```
Comment (aortizoj15) [inline src/api/admin.ts:170]: remove console.log

File: src/public/providers.ts [hunk 3]
```diff
@@ -139,6 +139,7 @@ class ProviderKeyElement {
         providerKeyElement
       ) as HTMLInputElement
       const method = isNew ? 'POST' : 'PATCH'
+      // return console.log('click', providerKey, method)
```
Comment (aortizoj15) [inline src/public/providers.ts:142]: is this still needed?

File: src/api/admin.ts (hunk 2 above)
Comment (coltonehrman) [reply src/api/admin.ts:170]: Good catches :)

File: src/providers/namecheap.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    namecheapApi.config.set('ApiUser', findKey(keys[0]))
+    namecheapApi.config.set('ApiKey', findKey(keys[1]))
+    namecheapApi.config.set('ClientIp', await ifconfigRes.text())
+
+    const { response } = await namecheapApi.apiCall(
+      'namecheap.domains.getList',
+      {}
+    )
+
+    response[0].DomainGetListResult[0].Domain.forEach(domain => {
```
Comment (songz) [inline src/providers/namecheap.ts:41]: `_.get(` -> lodash

File: src/providers/namecheap.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  const keysDefault: { key: string }[] = [{ key: keys[0] }, { key: keys[1] }]
+  const providerKeys = keysDefault.map(key => {
+    const serviceKeys = getProviderKeys()
+    return serviceKeys.find(k => k.service === dns && k.key === key.key) || key
+  })
+  return providerKeys as ServiceKey[]
+}
+const findKey = (key: string): string => {
+  return (getKeys().find(k => k.key === key) || { value: '' }).value
+}
```
Comment (songz) [inline src/providers/namecheap.ts:23]: could you turn `getKeys` and `findKeys` into a helper function to make the provider code simpler? Will help make future integrations simpler. `/lib/provider.ts`

Comment (coltonehrman) [reply src/providers/namecheap.ts]: i missed the part about `/lib/provider.ts` lol but check out how i did it, and let me know what you think :)

**Pull request merged**